distributed/scheduler.py

```diff
@@ -3124,7 +3124,9 @@ async def rebalance(self, comm=None, keys=None, workers=None):
                 if not all(r["status"] == "OK" for r in result):
                     return {
                         "status": "missing-data",
-                        "keys": tuple(concat(r["keys"].keys() for r in result)),
+                        "keys": tuple(
+                            concat(r["keys"].keys() for r in result if "keys" in r)
```

Comment:
```suggestion
                            concat(r["keys"].keys() for r in result if r["status"] == "missing-data")
```

This seems like it might be more explicit / easier for future maintainers to understand what is going on